main.js

```diff
@@ -81,11 +82,16 @@ define(function (require, exports, module) {
                 if (stream.sol()) {
                     _isLeading  = true;
                     _isTrailing = false;
+                    _isEmptyLine = false;
                     
                     _trailingOffset = stream.string.length;
                     trailing = stream.string.match(/[ \t]+$/);
                     if (trailing) {
                         _trailingOffset -= trailing[0].length;
+                        // everything is whitespace
+                        if(_trailingOffset == 0) {
```

Comment:
Small nits here (nothing major): Capitalize "everything", one space between `if` and `(`, and use strict equality (`===`).